### PR TITLE
Add buckets for service sessions

### PR DESCRIPTION
**Pull request by bmuniswamy**
# Critical Changes

# Changes
Created a new Bucketed Field metadata for service session status
Created 2 new buckets to represent the service session status. The two new buckets are Pending and Complete
Now the recent sessions component will be driven off of the bucket fields. Any status value under the Complete bucket will show up with a green checkmark on the component
If there is a status that is not part of any of the buckets then that status will not show up on the recent service session component
The attendance component is also driven off of the bucket values. Any status value under the Complete bucket will put the component in read only mode with an update button
Any status under the Pending bucket will put the attendance component in edit mode and the users will see the submit button.
# Issues Closed

# New Metadata

# Deleted Metadata

# Definition of Done
  Refer to [Asteroids DoD document](https://salesforce.quip.com/iq2mAy4i62oM) to see any additional details for the items below
~~- [ ] Any net new LWC work has Sa11y tests & 50% or above lines JEST test coverage~~  
~~- [ ] CRUD/FLS is enforced in Apex~~
~~- [ ] Permission sets are updated to account for CRUD|FLS|Tab|Classes~~
~~- [ ] Field sets are updated to account for new fields~~
~~- [ ] UX approval or UX not necessary~~
- [x] Link the pull request and work item by PR comment and Chatter post respectively, e.g. GUS: W-0000000: Work Name (https://github.com/SalesforceFoundation/PMM/pull/303)
- [ ] All **acceptance criteria** have been met
    - [x] Developer
    - [ ] Code Reviewer
    - [ ] QA
- [x] PR contains draft release notes
- [ ] QE story level testing completed

### CODE REVIEW

**Comment by bmuniswamy** [comment]
[W-9655434](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07AH000000SBVxYAO/view)

File: force-app/main/default/classes/AttendanceController.cls
Comment (ptewson-sfdo) [inline force-app/main/default/classes/AttendanceController.cls:44]: What do you think about defining the bucketNames, objectType, and bucketedField as properties of the AttendanceController instead of passing them in as arguments? The name of this method suggests it only returns buckets for the status field on ServiceSession, but the way it's coded the caller can actually get bucket info for any field an object. 

If the bucketNames, objectType, and bucketedField are defined by the AttendanceController class instead of passed in from the javascript, I think the interface is a little big cleaner.

Comment (ptewson-sfdo) [reply force-app/main/default/classes/AttendanceController.cls:44]: (Or just defining those values in the method itself probably works too)

File: force-app/main/default/classes/ServiceSessionService.cls
Comment (ptewson-sfdo) [inline force-app/main/default/classes/ServiceSessionService.cls:45]: Following up from my comment above, I think this method should be renamed to something like "getServiceSessionStatusesForBuckets" and the argument being passed in should be something like "bucketName". While the returned values look to be Service Session status values, what is being passed in appears to be the name of the bucket, not the status (see line #46 directly below this)

File: force-app/main/default/classes/ServiceDeliveryService.cls
Comment (ptewson-sfdo) [inline force-app/main/default/classes/ServiceDeliveryService.cls:246]: Should remove these debug logs

File: force-app/main/default/classes/ServiceDeliveryService.cls
Comment (ptewson-sfdo) [inline force-app/main/default/classes/ServiceDeliveryService.cls:240]: I'm not sure the updates to this class are necessary - I think they could reverted without altering the rest of the delivery here. I also prefer the restricted argument list to "getStatusBuckets" because I think that the ServiceDeliveryService should really be responsible for delivering bucket info for the ServiceDelivery sobject only, and the method all already narrows the scope to the status field.

File: force-app/main/default/classes/ServiceSessionSelector.cls
Comment (ptewson-sfdo) [inline force-app/main/default/classes/ServiceSessionSelector.cls:59]: This is another method that I would suggest gets a re-name... it's returning statuses, not buckets

File: force-app/main/default/classes/ServiceSessionSelector.cls
Comment (ptewson-sfdo) [inline force-app/main/default/classes/ServiceSessionSelector.cls:50]: Should clear out this debug (and on line 53) before merging in?

File: force-app/main/default/classes/ServiceSessionSelector.cls
Comment (ptewson-sfdo) [reply force-app/main/default/classes/ServiceSessionSelector.cls:59]: And actually - Jenny should maybe have the final word here - but I don't know if this method needs to exist in the ServiceSessionSelector. The way I think of it, this method belongs more in the ServiceSessionService - or even the controller, because the logic coded in here (returning status values for the two buckets that get displayed on the home view) is more specific to the front end. I don't think there needs to be a separate selector method for this behavior, just calling straight through to the FieldBucketSelector from the ServiceSessionService....

Comment (ptewson-sfdo) [reply force-app/main/default/classes/ServiceSessionSelector.cls:57]: @bmuniswamy I still believe this method needs to be renamed to reflect that it's returning a set of status values, not buckets - I would go with "getAllowedServiceSessionStatusValues()" maybe? And then update the line where you are calling it (line #18?) with the new name

**Review by ptewson-sfdo — APPROVED** [review]
Looks great, nice work!

**Pull request merged**
